xblock/mixins.py

```diff
@@ -522,7 +522,9 @@ def _add_field(self, node, field_name, field):
             elem.text = value
             node.insert(0, elem)
         else:
-            node.set(field_name, value)
+            # setting the None in the value for node raises TypeError to avoid this
```

Comment:
For OLX, if the value is the default for the field, then it shouldn't be set. Otherwise, it should be serialized to a string by the field. It looks like https://github.com/edx/XBlock/blob/master/xblock/mixins.py#L477-L478 takes care of not adding the field if it's a default. `field.to_string` should take care of the serialization. In what case is `field.to_string` returning `None`?